src/sync_template.sh

```diff
@@ -243,11 +251,25 @@ function pull_source_changes() {
 
   eval "git pull ${source_repo} --tags ${git_remote_pull_params}" || pull_has_issues=true
 
+  info "finished pulling from the source."
+  info "logging out from source ${SOURCE_REPO_HOSTNAME}."
+
+  if [[ -n "${SRC_SSH_PRIVATEKEY_ABS_PATH}" ]] &>/dev/null; then
+    info "we are using ssh for the source repo. No need to logout."
+  else
+    gh auth logout --hostname "${SOURCE_REPO_HOSTNAME}"
+
+    info "logging in into the target with hostname ${GITHUB_SERVER_URL}"
+    gh auth login --git-protocol "https" --hostname "${GITHUB_SERVER_URL}"
```

Comment:
Does this use the token of the pipeline? This was kind of my assumption that the push would use the pipeline token and not the generated token for the source repo. Though I missed the logout rightfully in my PR

Reply:
Yes it is using the token of the pipeline.
Am not super sure if after a logout is is still using the right token env variable. That's why it is possibly good to have it explicit. But I had the same thought if it is needed.